games/hangman.c

```diff
@@ -0,0 +1,291 @@
+    while ((strchr(game.hidden, '_') != NULL) && game.incorrect <= 12) {
+        do {
+            printf("\n****************************\n");
+            printf("Your word: ");
+
+            for (int i = 0; i < game.size; i++) {
+                printf("%c ", game.hidden[i]);
+            }
+
+            if (game.guesses_size > 0) {
+                printf("\nSo far, you have guessed: ");
+                for (int i = 0; i < game.guesses_size; i++) {
+                    printf("%c ", game.guesses[i]);
+                }
+            }
+
+            printf("\nYou have %d guesses left.", (12 - game.incorrect));
+            printf("\nPlease enter a letter: ");
+            scanf(" %c", &guess);
+            guess = tolower(guess);
+
+        } while (new_guess(guess, game.guesses, game.guesses_size) != -1);
+
+        game.guesses[game.guesses_size] = guess; // adds new letter to guesses array
+        game.guesses_size++; // updates size of guesses array
+
+        if (in_word(guess, game.current_word, game.size) == 1) {
+            printf("That letter is in the word!");
+            for (int i = 0; i < game.size; i++) {
+                if ((game.current_word[i]) == guess) {
+                    game.hidden[i] = guess;
+                }
+            }
+        } else {
+            printf("That letter is not in the word.\n");
+            (game.incorrect)++;
+        }
+        picture(game.incorrect);
+    }
+
+    won(game.current_word, game.incorrect);
+    return 0;
+}
+
+/**
+ * @brief checks if letter has been guessed before
+ * @param new_guess letter that has been guessed by player
+ * @param guesses array of player's previous guesses
+ * @param size size of guesses[] array
+ * @returns 1 if letter has been guessed before
+ * @returns -1 if letter has not been guessed before
+ */
+int new_guess(char new_guess, const char guesses[], int size) {
+
+    for (int j = 0; j < size; j++) {
+        if (guesses[j] == new_guess) {
+            printf("\nYou have already guessed that letter.");
+            return 1;
+        }
+    }
+
+    return -1;
+}
+
+/**
+ * @brief checks if letter is in current word
+ * @param letter letter guessed by player
+ * @param word current word
+ * @param size length of word
+ * @returns 1 if letter is in word
+ * @returns -1 if letter is not in word
+ */
+int in_word(char letter, const char word[], unsigned int size) {
+
+    for (int i = 0; i < size; i++) {
+        if ((word[i]) == letter) {
+            return 1;
+        }
+    }
+
+    return -1;
+}
+
+/**
+ * @brief creates a new game - generates a random word and stores in global variable current_word
+ * @returns current_game - a new game instance containing randomly selected word, its length and hidden version of word
+ */
+struct game_instance new_game() {
+
+    char word[30]; // used throughout function
+
+    FILE *fptr;
+    fptr = fopen("words.txt", "r");
+
+    if (fptr == NULL){
+        fprintf(stderr, "File not found.");
```

Comment:
Probably a file should be provided to test the game itself.
```suggestion
        fprintf(stderr, "File not found.\n");
```